File: AndroidCustomToolsFromLorenWang/src/main/java/android/lorenwang/tools/app/SharedPrefUtils.java
package android.lorenwang.tools.app;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.Set;

/**
 * 创建时间：2019-04-04 下午 17:29:20
 * 创建人：王亮（Loren wang）
 * 功能作用：Android prefence文件读写操作工具类
 * 思路：
 * 方法：
 * 注意：
 * 修改人：
 * 修改时间：
 * 备注：
 */

public class SharedPrefUtils {
    private static SharedPrefUtils sharedPrefUtils;
    private SharedPreferences mPref;

    public SharedPrefUtils(Context context) {
        mPref = PreferenceManager
                .getDefaultSharedPreferences(context.getApplicationContext());
    }

    public static SharedPrefUtils getInstance(Context context) {
        if (sharedPrefUtils == null && context != null) {
            sharedPrefUtils = new SharedPrefUtils(context);
        }
        return sharedPrefUtils;
    }


    public SharedPreferences getSharedPreferences(Context context, String name) {
        return context.getApplicationContext().getSharedPreferences(name, Context.MODE_PRIVATE);
    }

    public SharedPreferences getSharedPreferences(Context context, String name, int mode) {
        return context.getApplicationContext().getSharedPreferences(name, mode);
    }

    public Boolean clear() {
        return clear(mPref);
    }

    public Boolean clear(SharedPreferences pref) {
        SharedPreferences.Editor editor = pref.edit();
        editor.clear();
        return editor.commit();
    }

    public Boolean remove(String key) {
        return remove(mPref, key);
    }


    public Boolean remove(SharedPreferences pref, String key) {
        SharedPreferences.Editor editor = pref.edit();
        editor.remove(key);
        return editor.commit();
    }

    public Boolean contains(String key) {
        return contains(mPref, key);
    }

    public Boolean contains(SharedPreferences pref, String key) {
        return pref.contains(key);
    }

    /*--------------------------------------------------------------------------
	| 读数据
	--------------------------------------------------------------------------*/
    public int getInt(String key, int defValue) {
        return getInt(mPref, key, defValue);
    }


    public int getInt(SharedPreferences pref, String key, int defValue) {
        return pref.getInt(key, defValue);
    }

    public Long getLong(String key, Long defValue) {
        return getLong(mPref, key, defValue);
    }


    public Long getLong(SharedPreferences pref, String key, Long defValue) {
        return pref.getLong(key, defValue);
    }

    public String getString(String key, String defValue) {
        return getString(mPref, key, defValue);
    }

    public Set<String> getStringSet(String key, Set<String> defValue) {
        return getStringSet(mPref, key, defValue);
    }


    public String getString(SharedPreferences pref, String key, String defValue) {
        return pref.getString(key, defValue);
    }

    public Set<String> getStringSet(SharedPreferences pref, String key, Set<String> defValue) {
        return pref.getStringSet(key, defValue);
    }

    public Boolean getBoolean(String key, Boolean defValue) {
        return getBoolean(mPref, key, defValue);
    }


    public Boolean getBoolean(SharedPreferences pref, String key, Boolean defValue) {
        return pref.getBoolean(key, defValue);
    }

    /*--------------------------------------------------------------------------
	| 写数据
	--------------------------------------------------------------------------*/
    public Boolean putInt(String key, int value) {
        return putInt(mPref, key, value);
    }


    public Boolean putInt(SharedPreferences pref, String key, int value) {
        SharedPreferences.Editor editor = pref.edit();
        editor.putInt(key, value);
        return editor.commit();
    }

    public Boolean putLong(String key, Long value) {
        return putLong(mPref, key, value);
    }


    public Boolean putLong(SharedPreferences pref, String key, Long value) {
        SharedPreferences.Editor editor = pref.edit();
        editor.putLong(key, value);
        return editor.commit();
    }

    public Boolean putString(String key, String value) {
        return putString(mPref, key, value);
    }

    public Boolean putString(String key, Set<String> value) {
        return putStringSet(mPref, key, value);
    }

    public Boolean putString(SharedPreferences pref, String key, String value) {
        SharedPreferences.Editor editor = pref.edit();
        editor.putString(key, value);
        return editor.commit();
    }

    public Boolean putStringSet(SharedPreferences pref, String key, Set<String> value) {
        SharedPreferences.Editor editor = pref.edit();
        editor.putStringSet(key, value);
        return editor.commit();
    }

    public Boolean putBoolean(String key, Boolean value) {
        return putBoolean(mPref, key, value);
    }

    public Boolean putBoolean(SharedPreferences pref, String key, Boolean value) {
        SharedPreferences.Editor editor = pref.edit();
        editor.putBoolean(key, value);
        return editor.commit();
    }

}

File: AndroidCustomToolsFromLorenWang/src/main/java/android/lorenwang/tools/app/SharedPrefUtils.kt
package android.lorenwang.tools.app

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager


/**
 * Android prefence文件读写操作工具类
 *
 * @author yynie
 */
class SharedPrefUtils(context: Context) {

    private val mPref: SharedPreferences

    init {
        mPref = PreferenceManager
                .getDefaultSharedPreferences(context.applicationContext)
    }

    companion object {
        private var baseUtils: SharedPrefUtils? = null

        fun getInstance(context: Context):SharedPrefUtils{
            if(baseUtils == null){
                baseUtils = SharedPrefUtils(context)
            }
            return baseUtils as SharedPrefUtils;
        }
    }


    fun getSharedPreferences(context: Context?, name: String): SharedPreferences? {
        return context?.applicationContext?.getSharedPreferences(name,
                Context.MODE_PRIVATE)
    }

    fun getSharedPreferences(context: Context?, name: String, mode: Int): SharedPreferences? {
        return context?.applicationContext?.getSharedPreferences(name, mode)
    }

    fun clear(): Boolean {
        return clear(mPref)
    }

    fun clear(pref: SharedPreferences = mPref): Boolean {
        val editor = pref.edit()
        editor.clear()
        return editor.commit()
    }

    fun remove(key: String): Boolean {
        return remove(mPref, key)
    }


    fun remove(pref: SharedPreferences, key: String): Boolean {
        val editor = pref.edit()
        editor.remove(key)
        return editor.commit()
    }

    operator fun contains(key: String): Boolean {
        return contains(mPref, key)
    }

    fun contains(pref: SharedPreferences, key: String): Boolean {
        return pref.contains(key)
    }

    /*--------------------------------------------------------------------------
	| 读数据
	--------------------------------------------------------------------------*/
    fun getInt(key: String, defValue: Int): Int {
        return getInt(mPref, key, defValue)
    }


    fun getInt(pref: SharedPreferences, key: String, defValue: Int): Int {
        return pref.getInt(key, defValue)
    }

    fun getLong(key: String, defValue: Long): Long {
        return getLong(mPref, key, defValue)
    }


    fun getLong(pref: SharedPreferences, key: String, defValue: Long): Long {
        return pref.getLong(key, defValue)
    }

    fun getString(key: String, defValue: String?): String? {
        return getString(mPref, key, defValue)
    }

    fun getStringSet(key: String, defValue: Set<String>?): Set<String>? {
        return getStringSet(mPref, key, defValue)
    }


    fun getString(pref: SharedPreferences, key: String,
                  defValue: String?): String? {
        return pref.getString(key, defValue)
    }

    fun getStringSet(pref: SharedPreferences, key: String,
                     defValue: Set<String>?): Set<String>? {
        return pref.getStringSet(key, defValue)
    }

    fun getBoolean(key: String, defValue: Boolean): Boolean {
        return getBoolean(mPref, key, defValue)
    }


    fun getBoolean(pref: SharedPreferences, key: String,
                   defValue: Boolean): Boolean {
        return pref.getBoolean(key, defValue)
    }

    /*--------------------------------------------------------------------------
	| 写数据
	--------------------------------------------------------------------------*/
    fun putInt(key: String, value: Int): Boolean {
        return putInt(mPref, key, value)
    }


    fun putInt(pref: SharedPreferences, key: String, value: Int): Boolean {
        val editor = pref.edit()
        editor.putInt(key, value)
        return editor.commit()
    }

    fun putLong(key: String, value: Long): Boolean {
        return putLong(mPref, key, value)
    }


    fun putLong(pref: SharedPreferences, key: String, value: Long): Boolean {
        val editor = pref.edit()
        editor.putLong(key, value)
        return editor.commit()
    }

    fun putString(key: String, value: String?): Boolean {
        return putString(mPref, key, value)
    }

    fun putString(key: String, value: Set<String>?): Boolean {
        return putStringSet(mPref, key, value)
    }

    fun putString(pref: SharedPreferences, key: String,
                  value: String?): Boolean {
        val editor = pref.edit()
        editor.putString(key, value)
        return editor.commit()
    }

    fun putStringSet(pref: SharedPreferences, key: String,
                     value: Set<String>?): Boolean {
        val editor = pref.edit()
        editor.putStringSet(key, value)
        return editor.commit()
    }

    fun putBoolean(key: String, value: Boolean): Boolean {
        return putBoolean(mPref, key, value)
    }

    fun putBoolean(pref: SharedPreferences, key: String,
                   value: Boolean): Boolean {
        val editor = pref.edit()
        editor.putBoolean(key, value)
        return editor.commit()
    }


}
